src/React.Router/HtmlHelperExtensions.cs

```diff
@@ -0,0 +1,167 @@
+﻿/*
+ *  Copyright (c) 2014-Present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+using System;
+using React.Exceptions;
+using React.TinyIoC;
+
+#if NET451
+using System.Web;
+using System.Web.Mvc;
+using HttpResponse = System.Web.HttpResponseBase;
+using IHtmlHelper = System.Web.Mvc.HtmlHelper;
+#else
+using Microsoft.AspNetCore.Mvc.Rendering;
+using IHtmlString = Microsoft.AspNetCore.Html.IHtmlContent;
+using HttpResponse = Microsoft.AspNetCore.Http.HttpResponse;
+using Microsoft.AspNetCore.Html;
+#endif
+
+namespace React.Router
+{
+	/// <summary>
+	/// Render a React StaticRouter Component with context.
+	/// </summary>
+	public static class HtmlHelperExtensions
+	{
+		/// <summary>
+		/// Gets the React environment
+		/// </summary>
+		private static IReactEnvironment Environment
+		{
+			get
+			{
+				try
+				{
+					return ReactEnvironment.Current;
+				}
+				catch (TinyIoCResolutionException ex)
+				{
+					throw new ReactNotInitialisedException(
+#if NET451
+						"ReactJS.NET has not been initialised correctly.",
+#else
+						"ReactJS.NET has not been initialised correctly. Please ensure you have " +
+						"called services.AddReact() and app.UseReact() in your Startup.cs file.",
+#endif
+						ex
+					);
+				}
+			}
+		}
```

Comment:
In that case, I wanted to show a more specific error message for if ReactJS.NET was misconfigured. I guess that's useful here too though! Maybe `ReactEnvironment` should have a getter that does that (like `ReactEnvironment.GetCurrentOrThrow()`) so that the code is not duplicated.